neural_modelling/src/common/neuron-typedefs.h

```diff
@@ -51,56 +47,46 @@ union _spike_t {
         key_t key;
     };
 };
+#else  /*SPIKES_WITHOUT_PAYLOADS*/
+//! The type of a spike
+typedef uint32_t spike_t;
+#endif /*SPIKES_WITH_PAYLOADS*/
 
-//! \brief helper method to retrieve the key from a spike
+//! \brief Retrieve the key from a spike.
 //! \param[in] s: the spike to get the key from
-//! \return key_t: the key from the spike
+//! \return the key from the spike
 static inline key_t spike_key(spike_t s) {
```

Comment:
The key change I'm doing here is I'm trying to make it so that `spike_key` and `spike_payload` are only defined once; the differences are in the types they talk about and the body of the function. This makes the documentation process much nicer.